src/lib.rs

```diff
@@ -432,6 +436,42 @@ struct Chan<T> {
 }
 
 impl<T> Chan<T> {
+    #[inline(always)]
```

Comment:
I included this line to ensure there isn't a performance degradation from extracting out this function from the existing code.